pkg/compiler/codegen.go

```diff
@@ -234,9 +230,16 @@ func (c *codegen) emitDefault(n ast.Expr) {
 		default:
 			emit.Opcode(c.prog.BinWriter, opcode.PUSHNULL)
 		}
-		return
+	case *types.Slice:
+		if isCompoundSlice(t) {
+			emit.Opcode(c.prog.BinWriter, opcode.NEWARRAY0)
```

Comment:
```
var s []int
if s == nil {
    return 3
}
return 2
```
Will it work the way it's expected to?

Reply:
Maybe we can postpone it and fix later as a part of #949.

Reply:
Actually, `NullItem`'s can't be used as `Array` or `Struct`, so it is not much simpler to implement `== nil` comparisons in `NEO3`.